terraform/templates/lambda/reference_generator_policy.json.tpl

```diff
@@ -0,0 +1,32 @@
+{
```

Comment:
The Lambda will also need permissions to the KMS key that encrypts the table.

I think the following statement is what is needed:
...
```
{
      "Effect": "Allow",
      "Action": [
        "kms:GenerateDataKey",
        "kms:Decrypt",
         "kms:Decrypt"
      ],
      "Resource": "${kms_key_arn}"
    },
...
```
There is going to be a bit of trial an error with these permissions, but this is a start.

The value of the `kms_key_arn` param will be the KMS Key Arn as per here: https://github.com/nationalarchives/da-reference-generator/blob/main/terraform/root_main.tf#L12

In addition the KMS Key needs to give the Lambda role permission to use it. So the Terraform here: https://github.com/nationalarchives/da-reference-generator/blob/main/terraform/root_main.tf#L20 will need updating to this:
```
...
   user_roles    = [module.reference_generator_lambda.lambda_role_arn]
...
```